### PR TITLE
Change detection for single commits

### PR DESCRIPTION
**Pull request by mowies**
This PR fixes issue #40

### CODE REVIEW

**Comment by mowies** [comment]
I am not sure if this is nicely unit testable. Any ideas @ksindi ?

File: buildpipe/schema.json [hunk 1]
```diff
@@ -121,6 +121,11 @@
     }
   },
   "properties": {
+    "last_commit_only": {
+      "type": "boolean",
+      "description": "Gloabal setting - Check either only the paths from the last commit or the paths from the complete history (Default: false - complete history)",
```
Comment (ksindi) [inline buildpipe/schema.json]:
```suggestion
      "description": "Global setting - Check either only the paths from the last commit or the paths from the complete history (Default: false - complete history)",
```

File: buildpipe/pipeline.py [hunk 2]
```diff
@@ -62,12 +62,15 @@ def get_deploy_branch(config: box.Box) -> str:
...
+def get_changed_files(branch: str, deploy_branch: str, last_commit_only: bool) -> Set[str]:
     commit = os.getenv('BUILDKITE_COMMIT') or branch
     if branch == deploy_branch:
         command = ['git', 'log', '-m', '-1', '--name-only', '--pretty=format:', commit]
     else:
-        command = ['git', 'whatchanged', '--name-only', '--pretty=format:', 'origin..HEAD']
+        if last_commit_only:
+            command = ['git', 'log', '-1', '--name-only', '--no-merges', '--pretty=format:', 'origin..HEAD']
+        else:
+            command = ['git', 'log', '--name-only', '--no-merges', '--pretty=format:', 'origin..HEAD']
```
Comment (ksindi) [inline buildpipe/pipeline.py]: Thanks for this. Been meaning to change from `whatchanged`.

File: README.rst [hunk 3]
```diff
@@ -181,6 +181,23 @@ In the above config, if only files under `pyproject` were touched and the merge
...
+
+    # Only check the last commit for new changes
+    last_commit_only: true
+
+    # trigger deploy steps on master during business hours
+    deploy:
+      branch: master
+      timezone: US/Eastern
+
+The :code:`last_commit_only` flag allows you to choose between change detection in the whole history or only in the last commit that happened.
```
Comment (ksindi) [inline README.rst]: Should we clarify that this only happens for the non-default branch?

**Comment by ksindi** [comment]
Thanks again @mowies! Left a couple comments. Unfortunately I'm not sure how to test this either.

File: buildpipe/pipeline.py (hunk 2 above)
Comment (mowies) [reply buildpipe/pipeline.py]: :)

File: README.rst (hunk 3 above)
Comment (mowies) [reply README.rst]: I am not sure about the differentiation between default branch and other branches anyways. Could you explain why this is there?
But yes I will update the readme

Comment (ksindi) [reply README.rst]: The "default" branch is the branch to be merged. I think you only want this behavior for the "PR branch".

Comment (mowies) [reply README.rst]: Yeah I think you're right. I added some clarification to the readme now

File: README.rst [hunk 4]
```diff
@@ -194,8 +194,12 @@ Additional Features
     deploy:
       branch: master
       timezone: US/Eastern
+    .
+    .
+    .
 
 The :code:`last_commit_only` flag allows you to choose between change detection in the whole history or only in the last commit that happened.
+This feature is only applied if the current branch is **not** the defined :code:`deploy` branch.
```
Comment (ksindi) [inline README.rst]:
```suggestion
This feature is only applied if the current branch is **not** the defined :code:`deploy` branch, which by default is `master`.
```

**Comment by ksindi** [comment]
Thanks @mowies. Let me have a quick try. If all works I'll merge and deploy as `0.5.0`.

**Comment by ksindi** [comment]
Thanks again for the feature @mowies. If I can entice you on doing more issues like me know :-).

**Comment by mowies** [comment]
Awesome!
I can look through the other issues but I have to see if I can make time. Maybe even work time since I use buildkite at work :)

**Pull request merged**